app/controllers/twilio.js

```diff
@@ -1,114 +1,87 @@
-
-                  } catch (e) { sendResponse(); }
-                } else { sendResponse(); }
-              }).catch(handleError);
-            }
-          }).catch(handleError);
-        }
-
-        // log message being received
-        let now = new Date(Date.now()).toISOString().split("T");
-        console.log("Message received from " + fromNumber + " on " + now[0] + " at " + now[1]);
-
-      }).catch(handleError);
+    // Send a blank response before proceeding with logic
+    res.send("<?xml version='1.0' encoding='UTF-8'?><Response></Response>");
+
+    let fromNumber = from.replace(/\D+/g, "");
+    if (fromNumber.length == 10) { 
+      from = "1" + from; 
+    }
+    let text = req.body.Body.replace(/["']/g, "").trim();
+    let MessageStatus = req.body.SmsStatus;
+    let MessageSID = req.body.MessageSid;
+
+    // Log IBM Sensitivity measures
+    SentimentAnalysis.logIBMSentimentAnalysis(req.body);
     
-    } catch (e) {
-      console.log("Error with SMS POST: ", e)
-      console.log(e.trace)
-    };
+    Twilio.processIncoming(fromNumber, text, MessageStatus, MessageSID)
+    .then((conversations) => {
+
+      let conversationIds = conversations.map((conversation) => {
+        return conversation.convid;
+      });
+
+      Conversations.findByIds(conversationIds)
+    }).then((conversations) => {
+
+      conversations.forEach((conversation) => {
+        let content;
+        let commId = conversation.message.comm;
+        let conversationId = conversation.convid;
+        let inboundMessages = conversation.messages.filter((message) => {
+          return message.inbound;
+        });
+        let outboundMessages = conversation.messages.filter((message) => {
+          return !message.inbound;
+        });
+
+        if (conversation.clid == null) {
+          // This is a new conversation that has been started from unknown number
+          if (inboundMessages.length == 1) {
+            content = `Sorry! This # is not registered; 
+                      Help us find you. Reply with your name 
+                      in the following format: FIRST M LAST.`;
+          } else {
+            content = `Thanks for the message. A support member
+                      will place this number with the correct
+                      case manager as soon as possible.`;
+          }
+
+        } else if (inboundMessages.length > 1) {
+            let lastInboundDate = inboundMessages[inboundMessages.length - 1].created;
+            let d1 = new Date(lastInboundDate);
+            let d2 = new Date();
+            let timeLapsed = Math.round((d2.getTime() - d1.getTime()) / (3600*1000));
+
+            // If it's been more than 4 hours let's communicate
+            if (timeLapsed > 1) {
+              let dayOfWeek = d2.getDay();
+              if (dayOfWeek == 0 || dayOfWeek == 6) {
+                content = `Message received. Because it 
+                          is the weekend, your case manager may not 
```

Comment:
As much as I love the multiline quotes I think the line breaks will be preserved in the sent text message here and look strange to the end user